### DISCUSSION TITLE
Is there any plan to develop zephyr to mircrokenrel architecture?

### DISCUSSION THREAD

**Discussion by wjhtinger** [zephyrproject-rtos/zephyr#43472] [Q&A]
Just like seL4, QNX etc.
In Safety Critical System, such as automotive, mircrokenrel have more and more attentions. And, is there any chance that  zephyr can replace QNX?

**Comment by nashif** [comment]
zephyr is not based on the microkernel architecture, however it does support userspace and thread isolation on MMU/MPU hardware. There are no plans to move to microkernel.

**Reply by wjhtinger** [reply]
> zephyr is not based on the microkernel architecture, however it does support userspace and thread isolation on MMU/MPU hardware. There are no plans to move to microkernel.

just that zephyr support userspace and thread isolation on MMU/MPU hardware, so if I only use zephyr core in kernel space and develop my device driver and midleware in user space, which can be said that zephyr is a microkernel in my project?
